massgov/mayflower
# React/DP-14439, DP-12214:  Header fixes.

## Description

- (React) [Header] DP-14439: Adds support for headerSearch prop to optionally be a function.
- (React) [Header] DP-14439: Ports code from Budget sites that fixes bugs related to missing classList functionality and browsers.
- (React) [MainNav] DP-12214: Updates markup to better handle keyboard/focus events. Also updates roles for a11y functionality.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-12214)
- [JIRA issue](https://jira.mass.gov/browse/DP-14439)

## Steps to Test


1. Pull the branch, start up storybook. Verify that no functionality related to the Header component has
changed. Verify that for the MainNav, the requested changes have been added.
